include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -160,18 +133,18 @@ typename MatType::elem_type CMAES<SelectionPolicyType>::Optimize(
     BaseMatType covLower;
     while (!arma::chol(covLower, C[idx0], "lower"))
       C[idx0].diag() += std::numeric_limits<ElemType>::epsilon();
-
-    for (size_t j = 0; j < lambda; ++j)
+    std::vector<BaseMatType> z(params.lambda, BaseMatType(iterate.n_rows, iterate.n_cols));
+    for (size_t j = 0; j < params.lambda; ++j)
     {
       if (iterate.n_rows > iterate.n_cols)
       {
-        pStep[idx(j)] = covLower *
-            arma::randn<BaseMatType>(iterate.n_rows, iterate.n_cols);
+        z[j] = arma::randn<BaseMatType>(iterate.n_rows, iterate.n_cols);
+        pStep[idx(j)] = covLower * z[j];
       }
       else
       {
-        pStep[idx(j)] = arma::randn<BaseMatType>(iterate.n_rows, iterate.n_cols)
-            * covLower;
+        z[j] = arma::randn<BaseMatType>(iterate.n_rows, iterate.n_cols);
+        pStep[idx(j)] = z[j] * covLower;
```

Comment:
Do we need the extra `z` vector? Or can we directly use `Step[idx(j)] = arma::randn<BaseMatType>(iterate.n_rows, iterate.n_cols) * covLower;` instead of splitting it up into two seperate operations.

Reply:
It is useful in https://github.com/mlpack/ensmallen/pull/346#discussion_r928143063

Reply:
I see, make sense.